synapse/lib/lmdbslab.py

```diff
@@ -329,32 +326,16 @@ def pop(self, lkey, db=None):
     def delete(self, lkey, val=None, db=None):
         return self._xact_action(self.delete, lmdb.Transaction.delete, lkey, val, db=db)
 
-    def put(self, lkey, lval, dupdata=False, db=None):
-        return self._xact_action(self.put, lmdb.Transaction.put, lkey, lval, dupdata=dupdata, db=db)
+    def put(self, lkey, lval, dupdata=False, overwrite=True, db=None):
+        return self._xact_action(self.put, lmdb.Transaction.put, lkey, lval, dupdata=dupdata, overwrite=overwrite,
```

Comment:
LMDB has an optional arg for every possible contingency :D

Reply:
indeed